### PR TITLE
Add '_' and '-' to ids validation

### PR DESCRIPTION
**Pull request by Jakub89**
We revert previous bugfix for this one.

### CODE REVIEW

File: concent_api/core/validation.py [hunk 1]
```diff
@@ -50,7 +52,8 @@ def validate_id_value(value, field_name):
             error_code=ErrorCode.MESSAGE_VALUE_WRONG_LENGTH,
         )
 
-    if not value.isalnum():
+    valid_id_regex = re.compile(r'^[a-zA-Z0-9_-]*')
```
Comment (rwrzesien) [inline concent_api/core/validation.py]: Please move this to constants. There is no need to compile this on each validation.

File: concent_api/core/tests/test_unit_validation.py [hunk 2]
```diff
@@ -112,8 +113,8 @@ def test_that_function_should_pass_when_value_is_alphanumeric(self):
     def test_that_function_should_raise_exception_when_value_is_not_alphanumeric(self):
```
Comment (dybi) [inline concent_api/core/tests/test_unit_validation.py]: should be renamed: `_` and `-` are accepted as well
EDIT:
please rename the first test too

Comment (dybi) [inline concent_api/core/tests/test_unit_validation.py]: BTW: Could you please make this class inherit from `unittest.TestCase`? There is no need for heavy `ConcentIntegrationTestCase` base.

File: concent_api/core/validation.py (hunk 1 above)
Comment (dybi) [inline concent_api/core/validation.py]: If you are using `fullmatch`, than I guess `^` is redundant

Comment (Jakub89) [reply concent_api/core/validation.py]: Done

File: concent_api/core/tests/test_unit_validation.py (hunk 2 above)
Comment (Jakub89) [reply concent_api/core/tests/test_unit_validation.py]: Done

Comment (Jakub89) [reply concent_api/core/tests/test_unit_validation.py]: Done

File: concent_api/core/validation.py (hunk 1 above)
Comment (Jakub89) [reply concent_api/core/validation.py]: Done

**Pull request merged**